src/middlewared/middlewared/plugins/vm.py

```diff
@@ -1643,10 +1924,13 @@ class VMFSAttachmentDelegate(FSAttachmentDelegate):
                 self.middleware.logger.warning('Unable to %s %r', action, attachment['id'])
 
 
-def setup(middleware):
+async def setup(middleware):
     global ZFS_ARC_MAX_INITIAL
     ZFS_ARC_MAX_INITIAL = sysctl.filter('vfs.zfs.arc.max')[0].value
     asyncio.ensure_future(kmod_load())
     asyncio.ensure_future(middleware.call('pool.dataset.register_attachment_delegate',
                                           VMFSAttachmentDelegate(middleware)))
+
+    if await middleware.call('system.ready'):
+        asyncio.ensure_future(middleware.call('vm.initialize_vms', 2))  # We use a short timeout here deliberately
```

Comment:
If middleware dies, what happens to VMs?

Reply:
VM's live on as they are being handled by libvirt. However VNC won't work anymore in that case as the subprocess would stop as well when middlewared dies

Reply:
So newly started middleware will see that VMs as already running?

Reply:
Yes

Reply:
Good, thank you